db/python/tables/sequencing_group.py

```diff
@@ -69,17 +76,108 @@ async def query(
+        )
+
+        if filter_.assay_meta is not None:
+            exclude_fields.append('assay_meta')
+            wheres, values = filter_.to_sql(sql_overrides, only=['assay_meta'])
+            query_values.update(values)
+            _query.append(
+                f"""
+            INNER JOIN (
+                SELECT DISTINCT
+                    sequencing_group_id
+                FROM
+                    sequencing_group_assay
+                    INNER JOIN (
+                        SELECT
+                            id
+                        FROM
+                            assay
+                        WHERE
+                            {wheres}
+                    ) AS assay_subquery ON sequencing_group_assay.assay_id = assay_subquery.id
+                ) AS sga_subquery ON sg.id = sga_subquery.sequencing_group_id
+            """
+            )
+
+        if filter_.created_on is not None:
+            exclude_fields.append('created_on')
+            wheres, values = filter_.to_sql(sql_overrides, only=['created_on'])
+            query_values.update(values)
+            _query.append(
+                f"""
+            INNER JOIN (
+                SELECT
+                    id,
+                    TIMESTAMP(min(row_start)) AS created_on
+                FROM
+                    sequencing_group FOR SYSTEM_TIME ALL
+                WHERE
+                    {wheres}
+                GROUP BY
+                    id
+            ) AS sg_timequery ON sg.id = sg_timequery.id
+            """
+            )
+
+        if filter_.has_cram is not None or filter_.has_gvcf is not None:
+            exclude_fields.extend(['has_cram', 'has_gvcf'])
+            wheres, values = filter_.to_sql(
+                sql_overrides, only=['has_cram', 'has_gvcf']
+            )
+            query_values.update(values)
+            _query.append(
+                f"""
+            INNER JOIN (
+                SELECT
+                    sequencing_group_id,
+                    FIND_IN_SET('cram', GROUP_CONCAT(LOWER(anlysis_query.type))) > 0 AS has_cram,
+                    FIND_IN_SET('gvcf', GROUP_CONCAT(LOWER(anlysis_query.type))) > 0 AS has_gvcf
+                FROM
+                    analysis_sequencing_group
+                    INNER JOIN (
+                        SELECT
+                            id, type
+                        FROM
+                            analysis
+                    ) AS anlysis_query ON analysis_sequencing_group.analysis_id = anlysis_query.id
+                GROUP BY
+                    sequencing_group_id
+                HAVING
+                    {wheres}
+            ) AS sg_filequery ON sg.id = sg_filequery.sequencing_group_id
+            """
+            )
+
+        # Add the rest of the filters
+        wheres, values = filter_.to_sql(sql_overrides, exclude=exclude_fields)
+        _query.append(
+            f"""
+            WHERE {wheres}"""
+        )
```

Comment:
This should be sufficient given the newline join below
```suggestion
        _query.append(f"WHERE {wheres}")
```